inc/hcupc-support.h

```diff
@@ -103,6 +103,17 @@ inline void execute_hcupc_lambda(T* lambda) {
 			 * If a lambda is heap allocated then the pointer will not be valid at
 			 * remote place. Due to this we have to first copy the content of this
 			 * lambda into a stack allocated lambda.
+			 *
+			 * XXX - This description doesn't make any sense to me.
+			 * There's no way in C++ to differentiate a pointer to
+			 * an object with dynamic vs automatic storage duration,
+			 * so I don't see how this could be true. To further
+			 * complicate this, we're currently using heap-allocated
+			 * stacks for our fibers, so everything is in the heap
+			 * (i.e., even address-range-checking wouldn't work for
+			 * differentiating between stack and heap). This will
+			 * need to be refactored later, but I need to find out
+			 * what exactly it's supposed to be doing first.
 			 */
 
 			/*
```

Comment:
It sounds like what was really needed here is a static assertion that the function object in question [`is_trivially_copyable`](http://en.cppreference.com/w/cpp/types/is_trivially_copyable#Notes). We might want to add such an assertion in the resource-worker-based HCUPC++ code in the future.
But yes, simply removing all of the HCUPC-specific code for now is probably easiest. (As you mentioned in the email, HCUPC++ users can instead use an older snapshot of HClib that is known to be compatible.)